pkg/knuu/instance_helper.go

```diff
@@ -124,6 +124,23 @@ func (i *Instance) deployPod() error {
 		return fmt.Errorf("failed to get image name: %v", err)
 	}
 
+	serviceAccountName := i.k8sName
+
+	// create a service account for the pod
+	if err := k8s.CreateServiceAccount(k8s.Namespace(), serviceAccountName, labels); err != nil {
+		return fmt.Errorf("failed to create service account: %v", err)
+	}
+
+	// create a role and role binding for the pod if there are policy rules
+	if len(i.policyRules) > 0 {
+		if err := k8s.CreateRole(k8s.Namespace(), i.k8sName, labels, i.policyRules); err != nil {
+			return fmt.Errorf("failed to create role: %v", err)
+		}
+		if err := k8s.CreateRoleBinding(k8s.Namespace(), i.k8sName, labels, serviceAccountName, i.k8sName); err != nil {
+			return fmt.Errorf("failed to create role binding: %v", err)
+		}
+	}
```

Comment:
I am confused. Why do we want to use k8sName as a serviceAccountName and remove instance's serviceAccountName variable?

Reply:
Reverted back in: https://github.com/celestiaorg/knuu/pull/129/commits/bbda78c6f4d51e1882a5b44d6aebfe73896dbac8